src/System.Private.CoreLib/shared/System/Environment.cs

```diff
@@ -18,6 +19,16 @@ public static partial class Environment
         /// </summary>
         internal static bool IsSingleProcessor => ProcessorCount == 1;
 
+        internal static int ProcessorIdModCount
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get
+            {
+                // As ProcessorCount will be a constant at Tier1; the Jit will drop the mod and use a faster approach.
+                return Thread.GetCurrentProcessorId() % ProcessorCount;
```

Comment:
Not sure if this just what `GetCurrentProcessorId` should return instead rather than a number potentially higher then `ProcessorCount`?